tests/testthat/test-build-articles.R

```diff
@@ -4,3 +4,40 @@ test_that("can recognise intro variants", {
   expect_true(article_is_intro("pack-age", "pack.age"))
   expect_true(article_is_intro("articles/pack-age", "pack.age"))
 })
+
+test_that("can build article that uses html_vignette", {
+  pkg <- as_pkgdown(test_path("assets/article-as-is"))
+  withr::defer(clean_site(pkg))
+
+  # theme is not set since html_vignette doesn't support it
+  expect_output(expect_error(build_article("html-vignette", pkg), NA))
+})
+
+test_that("can override html_document() options", {
+  pkg <- as_pkgdown(test_path("assets/article-as-is"))
+  withr::defer(clean_site(pkg))
+
+  expect_output(path <- build_article("html-document", pkg))
+
+  # Check that number_sections is respected
+  html <- xml2::read_html(path)
+  expect_equal(
+    html %>% xml2::xml_find_all(".//h1//span") %>% xml2::xml_text(),
+    c("1", "2"
+  ))
+})
+
+test_that("can override options with _output.yml", {
+  pkg <- as_pkgdown(test_path("assets/article-output"))
+  withr::defer(clean_site(pkg))
+
+  expect_output(path <- build_article("html-document", pkg))
+
+  # Check that number_sections is respected
+  html <- xml2::read_html(path)
+  expect_equal(
+    html %>% xml2::xml_find_all(".//h1//span") %>% xml2::xml_text(),
+    c("1", "2"
```

Comment:
same comment reg the parenthesis

Reply:
And now the student has become the master 😆